front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -14,10 +14,83 @@ import type {
   SimulationResponseSuccess,
 } from 'applications/operationalStudies/types';
 import type { ReportTrainV2 } from 'common/api/osrdEditoastApi';
+import { TAG_COLORS } from 'modules/simulationResult/consts';
+import type { SpeedLimitTagValue } from 'modules/simulationResult/types';
 import { mmToKm, msToKmh, mToKm } from 'utils/physics';
 
 import { electricalProfilesDesignValues } from './consts';
 
+const getTag = (source?: SpeedLimitTagValue['source']): { name: string; color: string } => {
+  let name: string;
+  let color: string;
+
+  switch (source?.speed_limit_source_type) {
+    case 'given_train_tag':
+      name = source.tag;
+      color = TAG_COLORS.GIVEN_TRAIN;
+      break;
+    case 'fallback_tag':
+      name = source.tag;
+      color = TAG_COLORS.FALLBACK;
+      break;
+    case 'unknown_tag':
+      name = 'incompatible';
+      color = TAG_COLORS.INCOMPATIBLE;
+      break;
+    default:
+      name = 'missing_from_train';
+      color = TAG_COLORS.MISSING;
+  }
+
+  return { name, color };
+};
+
+export const formatSpeedLimitTags = (
+  rawSpeedLimitTags: SimulationResponseSuccess['mrsp'],
+  pathLength: number
+): LayerData<SpeedLimitTagValues>[] => {
+  const mergedPositions: LayerData<SpeedLimitTagValues>[] = [];
+
+  rawSpeedLimitTags.values.forEach((value, index) => {
```

Comment:
How about using a `reduce` instead, or a real `for` loop? I find `.forEach` to have all the disadvantages: tries to be more 'functional programming' (so might be harder to read for those not used to it), but not fully embrace it.

```ts
return rawSpeedLimitTags.values.reduce([], (mergedPositions, value, index) => {
  /* same function */
});
```

or full `for` loop.
```ts
const mergedPositions: LayerData<SpeedLimitTagValues>[] = [];
for (const [value, index] of rawSpeedLimitTags.values.entries) {
  /* same function */
}
return mergedPositions;
```

Reply:
I decided to use the reduce. The speedLimitTags that were blinking seem to be much better thank you 👍